src/components/engagement/engagement.service.ts

```diff
@@ -888,6 +893,46 @@ export class EngagementService {
       session
     )) as InternshipEngagement;
 
+    // take mentor, growthPlan, countryOfOrigin out of object to verify the 'simple' properties
+    const {
+      mentor,
+      growthPlan: growthPlanOnly,
+      countryOfOrigin,
+      ...objectNoExtra
+    } = object;
+
+    const realChanges = await this.db.getActualChanges(objectNoExtra, input);
+    await this.authorizationService.verifyCanEditChanges(
+      InternshipEngagement,
+      objectNoExtra,
+      realChanges
+    );
+    if (mentorId) {
+      await this.authorizationService.verifyCanEdit({
+        resource: InternshipEngagement,
+        baseNode: object,
+        prop: 'mentor',
+        propName: 'mentorId',
+        resourceName: 'Engagement',
```

Comment:
This is defines the first part of the path for the field. It should be lowercase so the field in the exception becomes `engagement.mentorId`